kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,850 @@
+        print('closing figure')
+
+
+    fig.canvas.mpl_connect('button_press_event', press)
+    fig.canvas.mpl_connect('button_release_event', release)
+    fig.canvas.mpl_connect('key_press_event', keypress)
+    fig.canvas.mpl_connect('key_release_event', keyup)
+    fig.canvas.mpl_connect('motion_notify_event', motionnotify)
+    fig.canvas.mpl_connect('resize_event', resize)
+    fig.canvas.mpl_connect('scroll_event', scroll)
+    fig.canvas.mpl_connect('figure_enter_event', figure_enter)
+    fig.canvas.mpl_connect('figure_leave_event', figure_leave)
+    fig.canvas.mpl_connect('close_event', close)
+
+'''
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+import os
+import scipy
+import matplotlib
+from matplotlib._pylab_helpers import Gcf
+from matplotlib.backend_bases import RendererBase, GraphicsContextBase,\
+    FigureManagerBase, FigureCanvasBase, NavigationToolbar2
+from matplotlib.figure import Figure
+from matplotlib.transforms import Bbox
+from matplotlib.backend_bases import ShowBase
+from matplotlib.mathtext import MathTextParser
+from matplotlib import rcParams
+from hashlib import md5
+from matplotlib import _png
+from matplotlib.font_manager import weight_as_number
+
+try:
+    import kivy
+except ImportError:
+    raise ImportError("this backend requires Kivy to be installed.")
+
+from kivy.app import App
+from kivy.graphics.texture import Texture
+from kivy.graphics import Rectangle
+from kivy.uix.widget import Widget
+from kivy.uix.label import Label
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.behaviors import FocusBehavior
+from kivy.uix.actionbar import ActionBar, ActionView, \
+                                ActionButton, ActionToggleButton, \
+                                ActionPrevious, ActionOverflow, ActionSeparator
+from kivy.base import EventLoop
+from kivy.core.text import Label as CoreLabel
+from kivy.graphics import Color, Line
+from kivy.graphics import Rotate, Translate
+from kivy.graphics.tesselator import Tesselator
+from kivy.graphics.context_instructions import PopMatrix, PushMatrix
+from kivy.logger import Logger
+from kivy.graphics import Mesh
+from kivy.resources import resource_find
+from kivy.uix.stencilview import StencilView
+from kivy.core.window import Window
+from kivy.uix.button import Button
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.relativelayout import RelativeLayout
+from kivy.uix.popup import Popup
+from kivy.properties import ObjectProperty
+from kivy.clock import Clock
+
+import numpy as np
+import io
+from _functools import partial
+from math import cos, sin, pi
+
+# try:
+#     kivy.require('1.9.0') # I would need to check which release of
+#     Kivy would be the best suitable.
+# except AttributeError:
+#     raise ImportError(
+#         "kivy version too old -- it must have require_version")
+
+_debug = True
```

Comment:
This should be removed. Instead, wherever you're printing debug messages use kivy logger instead with debug or info level etc.